testsuite/MDAnalysisTests/coordinates/test_chainreader.py

```diff
@@ -123,34 +126,98 @@ class TestChainReaderCommonDt(object):
+        universe = mda.Universe(GRO, [(PDB, 'pdb'), (XTC, 'xtc'), (TRR,
+                                                                   'trr')])
+        assert universe.trajectory.n_frames == 21
+        assert_equal(universe.trajectory.filenames, [PDB, XTC, TRR])
 
     def test_set_one_format_tuple(self):
         universe = mda.Universe(PSF, [(PDB_small, 'pdb'), DCD])
-        assert_equal(universe.trajectory.n_frames, 99)
+        assert universe.trajectory.n_frames == 99
 
     def test_set_all_formats(self):
         universe = mda.Universe(PSF, [PDB_small, PDB_closed], format='pdb')
-        assert_equal(universe.trajectory.n_frames, 2)
+        assert universe.trajectory.n_frames == 2
+
+
+class TestChainReaderContinuous(object):
+    # description of the frame patterns to test are in comments as lists
+    @pytest.fixture
+    def top(self):
+        return atom_gro
+
+    @pytest.fixture
+    def trajs(self):
+        return [atom_0, atom_1, atom_2]
+
+    # [0 1 2 3] [2 3 4 5 6] [5 6 7 8 9]
+    def test_length(self, top, trajs):
+        u = mda.Universe(top, trajs, continuous=True)
+        assert u.trajectory.n_frames == 10
+        for i, ts in enumerate(u.trajectory):
+            assert_almost_equal(i, ts.time)
+
+    # [5 6 7 8 9] [2 3 4 5 6] [0 1 2 3]
+    def test_reorder(self, top, trajs):
+        u = mda.Universe(top, trajs[::-1], continuous=True)
+        assert u.trajectory.n_frames == 10
+        for i, ts in enumerate(u.trajectory):
+            assert_almost_equal(i, ts.time)
+
+    # [0 1 2 3] [5 6 7 8 9]
+    def test_missing_frame(self, top, trajs):
+        with pytest.raises(RuntimeError):
+            mda.Universe(top, [trajs[0], trajs[-1]], continuous=True)
+
+    # [0] [0 1 2 3], [0] [0] [0 1 2 3], [0] [0] [0] [0 1 2 3]
+    @pytest.mark.parametrize('frame', ([atom_single_frame, ],
+                                       [atom_single_frame, ] * 2,
+                                       [atom_single_frame, ] * 3))
+    def test_exclude_frame(self, top, frame):
+        u = mda.Universe(top, frame + [atom_0_dcd, ], continuous=True)
+        assert u.trajectory.n_frames == 4
+        for i, ts in enumerate(u.trajectory):
+            assert_almost_equal(i, ts.time)
+
+
+    # [0 1 2 3] [1]
+    def test_more_excludes(self, top):
+        u = mda.Universe(top,  [single_frames[1], atom_0_dcd], continuous=True)
+        assert u.trajectory.n_frames == 4
+        for i, ts in enumerate(u.trajectory):
+            assert_almost_equal(i, ts.time)
+
+    # [0 1 2 3] [3]
+    def test_last_sinle(self, top):
+        u = mda.Universe(top,  [single_frames[3], atom_0_dcd], continuous=True)
+        assert u.trajectory.n_frames == 4
+        for i, ts in enumerate(u.trajectory):
+            assert_almost_equal(i, ts.time)
+        u.trajectory[3]
+        assert u.trajectory.active_reader == u.trajectory.readers[-1]
+
+    # [0 1 2 3] [2 3 4 5 6] [5 6 7 8 9] [4] [8]
+    def test_length_with_exclude(self, top, trajs):
```

Comment:
This one is another edge case I found that fails now. Does anyone have an idea how to get this to pass? It works with a single frame and just one other trajectory see the test on line 203.